src/lib/images/proxy.mjs

```diff
@@ -0,0 +1,124 @@
+import { Readable } from 'stream'
+
+import express from 'express'
+import { createIPX, ipxFSStorage, ipxHttpStorage, createIPXWebServer } from 'ipx'
+
+import { log, NETLIFYDEVERR } from '../../utils/command-helpers.mjs'
+
+export const parseAllDomains = function (config) {
+  const domains = config?.images?.remote_images
+  if (!domains) {
+    return { errors: [], remoteDomains: [] }
+  }
+
+  const remoteDomains = []
+  const errors = []
+
+  for (const patternString of domains) {
+    try {
+      const url = new URL(patternString)
+      if (url.hostname) {
+        remoteDomains.push(url.hostname)
+      } else {
+        errors.push(`The URL '${patternString}' does not have a valid hostname.`)
+      }
+    } catch (error) {
+      errors.push(`Invalid URL '${patternString}': ${error.message}`)
+    }
+  }
+
+  return { errors, remoteDomains }
+}
+
+const getErrorMessage = function ({ message }) {
+  return message
+}
+
+export const handleImageDomainsErrors = async function (errors) {
+  if (errors.length === 0) {
+    return
+  }
+
+  const errorMessage = await errors.map(getErrorMessage).join('\n\n')
+  log(NETLIFYDEVERR, `Image domains syntax errors:\n${errorMessage}`)
+}
+
+export const parseRemoteImageDomains = async function ({ config }) {
+  if (!config) {
+    return []
+  }
+
+  const { errors, remoteDomains } = await parseAllDomains(config)
+  await handleImageDomainsErrors(errors)
+
+  return remoteDomains
+}
+
+export const isImageRequest = function (req) {
+  const imageUrlPattern = /^\/\.netlify\/images/
+  return imageUrlPattern.test(req.url)
```

Comment:
nit: this can be solved without regex

```suggestion
const imageUrlPattern = "/.netlify/images/"
export const isImageRequest = function (req) {
  return req.url.startsWith(imageUrlPattern)
```